File: models/domain/Label.py
from flask_restful import fields


class Label():
    resource_fields = {
        'id': fields.Integer,
        'name': fields.String,
        'color': fields.String,
        'icon': fields.String,
    }

    def __init__(self, id=None, name=None, color=None, icon=None) -> None:
        self.id = id
        self.name = name
        self.color = color
        self.icon = icon

    def __repr__(self):
        return '<Label %r>' % self.name

File: frontend/src/app/components/transactions/label.ts
import { Category } from './category';

export class Label {
  public id: number;
  public name: string;
  public color: string;
  public icon: string;
  public category: Category;
  public numTransactions: number;

  constructor(id: number, name: string, color: string, icon: string, numTransactions: number) {
    this.id = id;
    this.name = name;
    this.color = color;
    this.icon = icon;
    this.numTransactions = numTransactions;
  }
}
